addons/web_editor/controllers/main.py

```diff
@@ -548,29 +548,46 @@ def subber(match):
             ('Cache-control', 'max-age=%s' % http.STATIC_CACHE_LONG),
         ])
 
-    @http.route(['/web_editor/media_library_url'], type='json', auth="user", website=True)
-    def media_library_url(self):
-        dbuuid = request.env['ir.config_parameter'].sudo().get_param('database.uuid')
+    @http.route(['/web_editor/media_library_search'], type='json', auth="user", website=True)
+    def media_library_search(self, **params):
         endpoint = request.env['ir.config_parameter'].sudo().get_param('web_editor.media_library_endpoint', DEFAULT_LIBRARY_ENDPOINT)
-        return '%s?dbuuid=%s' % (endpoint, dbuuid)
+        params['dbuuid'] = request.env['ir.config_parameter'].sudo().get_param('database.uuid')
+        response = requests.get('%s/1/search' % endpoint, params=url_encode(params))
+        if response.status_code == requests.codes.ok:
+            return response.json()
+        else:
+            return {'error': response.status_code}
 
-    @http.route('/web_editor/save_media', type='json', auth='user', methods=['POST'])
-    def save_media(self, media=[]):
+    @http.route('/web_editor/save_library_media', type='json', auth='user', methods=['POST'])
+    def save_library_media(self, media=[]):
         """
         Saves images from the media library as new attachments, making them
         dynamic SVGs if needed.
-            media = [
-                {
-                    'media_url': 'https://...',
+            media = {
+                <media_id>: {
                     'query': 'space separated search terms',
                     'is_dynamic_svg': True/False,
                 }, ...
-            ]
+            }
         """
         attachments = []
+        library_endpoint = request.env['ir.config_parameter'].sudo().get_param('web_editor.media_library_endpoint', DEFAULT_LIBRARY_ENDPOINT)
+
+        media_ids = ','.join([key for key, value in media])
+        params = {
+            'dbuuid': request.env['ir.config_parameter'].sudo().get_param('database.uuid'),
+            'media_ids': media_ids,
+        }
+        response = requests.get('%s/1/download_urls' % library_endpoint, params=url_encode(params))
+        if response.status_code != requests.codes.ok:
+            logger.exception("ERROR: couldn't get download urls from media library.")
+            raise Exception(_("ERROR: couldn't get download urls from media library."))
+        download_urls = json.loads(response.json())
+        for id, url in download_urls:
+            media[id]['download_url'] = url
 
         for medium in media:
```

Comment:
I love the latin, but since the medium have another meaning in marketing terms, can se simple use an other name ;)

btw since only tje download_urls is whitelisted, we should only create attachment for returned items.

So:
```python
for id, url in download_urls:
    req = requests.get(url)
    name = '_'.join([media[id]['query'], url.split('/')[-1]])
    attachment = request.env['ir.attachment'].create({ ... })
    if media[id].get('is_dynamic_svg''):
        attachment['url'] = '/web_editor/shape/external_shapes/%s' % slug(attachment)
```